pkg/kubelet/logs/container_log_manager.go

```
@@ -413,15 +413,19 @@ func (c *containerLogManager) removeExcessLogs(logs []string) ([]string, error)
 
 // compressLog compresses a log to log.gz with gzip.
 func (c *containerLogManager) compressLog(log string) error {
+	logInfo, err := os.Stat(log)
+	if err != nil {
+		return fmt.Errorf("failed to stat log file: %w", err)
```

can you add a variation for this new error return..

---

@mikebrow I'm a little confused what you mean by variation on this os.Stat call. Do you mean I should have a specific check if the file exists and then call os.Stat to get the logInfo separately? Such checks seem to be handled well in the following code. Perhaps you can elaborate with some pseudocode? Your guidance is appreciated.

---

I read this again and I wonder if you would like me to simply change the error string to be more descriptive and verbose...

---

Since the function compressLog only returns an error, I added a test to confirm compressing an unknown log should return an error in that case. Thanks for the feedback!!